xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -387,13 +403,19 @@ bool CDirectoryProvider::UpdateURL()
   if (value == m_currentUrl)
     return false;
 
+  if (URIUtils::IsProtocol(m_currentUrl, "pvr"))
+    g_PVRManager.UnregisterObserver(this);
+
   m_currentUrl = value;
 
   if (!m_isAnnounced)
   {
     m_isAnnounced = true;
     CAnnouncementManager::GetInstance().AddAnnouncer(this);
     ADDON::CAddonMgr::GetInstance().Events().Subscribe(this, &CDirectoryProvider::OnEvent);
+
+    if (URIUtils::IsProtocol(m_currentUrl, "pvr"))
```

Comment:
This wont work. m_currentUrl can change, and if it does, it wouldn't register correctly (just do the same as the two other registration systems)